spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -276,6 +277,13 @@ public void pruneColumns(StructType newRequestedSchema) {
 
   @Override
   public Statistics estimateStatistics() {
+    if (filterExpressions == null || filterExpressions == Expressions.alwaysTrue()) {
+      LOG.debug("using table metadata to estimate table statistics");
```

Comment:
Minor: I don't see much value in having this debug statement.

If you were trying to debug `estimateStatistics`, I think you'd want to have a message for every call, and you would want it to show the expression as well as the resulting estimate. I think Spark already has those logs, though.

Reply:
ack